src/wincrypt.rs

```diff
@@ -1417,6 +1417,260 @@ pub struct CTL_CONTEXT {
+pub const CERT_NAME_DNS_TYPE: ::DWORD = 6;
+pub const CERT_NAME_URL_TYPE: ::DWORD = 7;
+pub const CERT_NAME_UPN_TYPE: ::DWORD = 8;
+pub const CERT_SIMPLE_NAME_STR: ::DWORD = 1;
+pub const CERT_OID_NAME_STR: ::DWORD = 2;
+pub const CERT_X500_NAME_STR: ::DWORD = 3;
+
+pub const CERT_NAME_STR_SEMICOLON_FLAG: ::DWORD = 0x40000000;
+pub const CERT_NAME_STR_NO_PLUS_FLAG: ::DWORD = 0x20000000;
+pub const CERT_NAME_STR_NO_QUOTING_FLAG: ::DWORD = 0x10000000;
+pub const CERT_NAME_STR_CRLF_FLAG: ::DWORD = 0x08000000;
+pub const CERT_NAME_STR_COMMA_FLAG: ::DWORD = 0x04000000;
+pub const CERT_NAME_STR_REVERSE_FLAG: ::DWORD = 0x02000000;
+
+pub const CERT_NAME_ISSUER_FLAG: ::DWORD = 0x1;
+pub const CERT_NAME_STR_DISABLE_IE4_UTF8_FLAG: ::DWORD = 0x00010000;
+pub const CERT_NAME_STR_ENABLE_T61_UNICODE_FLAG: ::DWORD = 0x00020000;
+pub const CERT_NAME_STR_ENABLE_UTF8_UNICODE_FLAG: ::DWORD = 0x00040000;
+pub const CERT_NAME_STR_FORCE_UTF8_DIR_STR_FLAG: ::DWORD = 0x00080000;
+pub const CERT_DELETE_KEYSET_PROP_ID: ::DWORD = 101;
+pub const CERT_COMPARE_MASK: ::DWORD = 0xFFFF;
+pub const CERT_COMPARE_SHIFT: ::DWORD = 16;
+pub const CERT_COMPARE_ANY: ::DWORD = 0;
+pub const CERT_COMPARE_SHA1_HASH: ::DWORD = 1;
+pub const CERT_COMPARE_NAME: ::DWORD = 2;
+pub const CERT_COMPARE_ATTR: ::DWORD = 3;
+pub const CERT_COMPARE_MD5_HASH: ::DWORD = 4;
+pub const CERT_COMPARE_PROPERTY: ::DWORD = 5;
+pub const CERT_COMPARE_PUBLIC_KEY: ::DWORD = 6;
+pub const CERT_COMPARE_HASH: ::DWORD = CERT_COMPARE_SHA1_HASH;
+pub const CERT_COMPARE_NAME_STR_A: ::DWORD = 7;
+pub const CERT_COMPARE_NAME_STR_W: ::DWORD = 8;
+pub const CERT_COMPARE_KEY_SPEC: ::DWORD = 9;
+pub const CERT_COMPARE_ENHKEY_USAGE: ::DWORD = 10;
+pub const CERT_COMPARE_CTL_USAGE: ::DWORD = CERT_COMPARE_ENHKEY_USAGE;
+pub const CERT_COMPARE_SUBJECT_CERT: ::DWORD = 11;
+pub const CERT_COMPARE_ISSUER_OF: ::DWORD = 12;
+pub const CERT_COMPARE_EXISTING: ::DWORD = 13;
+pub const CERT_COMPARE_SIGNATURE_HASH: ::DWORD = 14;
+pub const CERT_COMPARE_KEY_IDENTIFIER: ::DWORD = 15;
+pub const CERT_COMPARE_CERT_ID: ::DWORD = 16;
+pub const CERT_COMPARE_CROSS_CERT_DIST_POINTS: ::DWORD = 17;
+pub const CERT_COMPARE_PUBKEY_MD5_HASH: ::DWORD = 18;
+pub const CERT_FIND_ANY: ::DWORD = CERT_COMPARE_ANY << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_SHA1_HASH: ::DWORD = CERT_COMPARE_SHA1_HASH << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_MD5_HASH: ::DWORD = CERT_COMPARE_MD5_HASH << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_SIGNATURE_HASH: ::DWORD = CERT_COMPARE_SIGNATURE_HASH << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_KEY_IDENTIFIER: ::DWORD = CERT_COMPARE_KEY_IDENTIFIER << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_HASH: ::DWORD = CERT_FIND_SHA1_HASH;
+pub const CERT_FIND_PROPERTY: ::DWORD = CERT_COMPARE_PROPERTY << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_PUBLIC_KEY: ::DWORD = CERT_COMPARE_PUBLIC_KEY << CERT_COMPARE_SHIFT;
+pub const CERT_FIND_SUBJECT_NAME: ::DWORD = CERT_COMPARE_NAME << CERT_COMPARE_SHIFT
+    | CERT_INFO_SUBJECT_FLAG;
```

Comment:
Perhaps parentheses would be useful here to clarify the order of operations, especially since this is coming from the world of C which has some rather non-conventional rules regarding << and |.